src/nvim/mbyte.c

```diff
@@ -118,6 +119,32 @@ struct interval {
 #endif
 
 /*
+ * To speed up BYTELEN() we fill a table with the byte lengths whenever
+ * enc_utf8 or enc_dbcs changes.
+ */
+static char mb_bytelen_tab[256];
+
+/*
+ * Return byte length of character that starts with byte "b".
+ * Returns 1 for a single-byte character.
+ * mb_byte2len_check() can be used to count a special key as one byte.
+ * Don't call mb_byte2len(b) with b < 0 or b > 255!
+ */
+char mb_byte2len(int b) {
+    assert(b >= 0);
+    assert(b <= 255);
```

Comment:
Also this function is pretty common. Are you sure asserts are needed? (Note: they AFAIK are not disabled even in release builds.)